include/ap/argument_parser.hpp

```diff
@@ -268,121 +487,186 @@ class optional_argument : public argument_interface {
 } // namespace detail
 
 
-
+/**
+ * @brief Represents a specifier for a positional argument.
+ * @details Provides a convenient way to specify a positional argument in the argument parser.
+ */
 struct positional {
     using type = detail::positional_argument;
 };
 
+/**
+ * @brief Represents a specifier for an optional argument.
+ * @details Provides a convenient way to specify an optional argument in the argument parser.
+ */
 struct optional {
     using type = detail::optional_argument;
 };
 
 namespace detail {
 
+/**
```

Comment:
I'm not sure whether it is necessary to have doc comments for the elements in the `detail` namespace. I also think that it may be better to move the concrete argument classes to a non-detail namespace so that only the utility elements are in the detail namespace. But that is a problem for another time